daemon/containerd/image_list.go

```diff
@@ -237,6 +238,77 @@ func (i *ImageService) imageSummary(ctx context.Context, img images.Image, platf
 			return nil
 		}
 
+		mfstSummary := imagetypes.ImageManifestSummary{
+			ID:         target.Digest.String(),
+			Available:  available,
+			Descriptor: target,
+			Kind:       imagetypes.ImageManifestKindUnknown,
+		}
+
+		if opts.Manifests {
+			defer func() {
+				// If the platform is available, prepend it to the list of platforms
+				// otherwise append it at the end.
+				if available {
+					manifestSummaries = append([]imagetypes.ImageManifestSummary{mfstSummary}, manifestSummaries...)
+				} else {
+					manifestSummaries = append(manifestSummaries, mfstSummary)
+				}
+			}()
+		}
+
+		contentSize, err := img.Size(ctx)
+		if err != nil {
+			if !cerrdefs.IsNotFound(err) {
+				log.G(ctx).WithFields(log.Fields{
+					"error":  err,
+					"image":  img.Name(),
+					"digest": target.Digest,
+				}).Warn("failed to determine size")
+			}
+		} else {
+			mfstSummary.Size.Content = contentSize
+			totalSize += contentSize
+			mfstSummary.Size.Total = totalSize
+		}
+
+		isPseudo, err := img.IsPseudoImage(ctx)
+		if err != nil && !errdefs.IsNotFound(err) {
+			log.G(ctx).WithFields(log.Fields{
+				"error":    err,
+				"image":    img.Name(),
+				"digest":   target.Digest,
+				"isPseudo": isPseudo,
+			}).Debug("pseudo image check failed")
+			return nil
```

Comment:
I notice that here we're returning `nil`, but for the `contentSize, err := img.Size(ctx)` error we're logging and continuing; is that intentional?

- Perhaps this line may need a comment why we don't (or can't) continue in this case. And why we _can_ continue if we get a `NotFound` error

Reply:
Yes, added a comment. Also the error checking below should also be ignored (so we don't abort the whole list operation just in case there's some issue with one image).